### PR TITLE
Factor out style-setting code

### PR DESCRIPTION
**Pull request by qfel**
# Description
This is mainly cleanup, but introduces a slight (positive, if anything) behavior change:

Some menu layouts support only a subset of styles, but with this change the user will still be able to configure them. This seems strictly better - if reedline starts supporting one of the existing styles for a particular layout, there won't be any need to update nushell code.

# User-Facing Changes
None

# Tests + Formatting
This is simply factoring out existing code, old tests should still cover it.

### CODE REVIEW

**Review by sholderbach — APPROVED** [review]
Thanks, sounds like a reasonable simplification to me.

As we haven't stress tested how different exotic menu configs behave should they now accept those options, we may hold off for a moment and land it after the upcoming release on Tuesday.

File: crates/nu-cli/src/reedline_config.rs
Comment (qfel) [reply crates/nu-cli/src/reedline_config.rs:196]: Do you have any suggestions? Maybe `menu_with_style`?

Comment (fdncred) [reply crates/nu-cli/src/reedline_config.rs:196]: yes, please make this change as to not confuse with builders and we can land it.

Comment (sholderbach) [reply crates/nu-cli/src/reedline_config.rs]: Went for a quick tweak to unblock this

**Pull request merged**
